File: Blind 75 LeetCode/Sakshi Rai/Best Time to Buy and Sell Stock/code.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = INT_MAX;
        int maxProfit = 0;

        int n = prices.size();
        if (n <= 1) {
            return 0; 
        }

        for (int i = 0; i < prices.size(); i++) {
            if (prices[i] < minPrice) {
                minPrice = prices[i]; 
            } else if (prices[i] - minPrice > maxProfit) {
                maxProfit = prices[i] - minPrice; 
            }
        }

        return maxProfit;
    }
};

File: Blind 75 LeetCode/Pankajavalli RV/Best Time to Buy and Sell Stock/code.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        lptr, profit = 0, 0
        for rptr in range(1, len(prices)):
            if prices[lptr] < prices[rptr]:
               profit = max(profit, prices[rptr] - prices[lptr])
            else:
                lptr = rptr
        return profit
